kore/src/Kore/Domain/Builtin.hs

```diff
@@ -236,6 +237,26 @@ emptyNormalizedAc = NormalizedAc
     , opaque = []
     }
 
+asSingleOpaqueElem
+    :: Eq (Element key child)
+    => Eq (Value key child)
+    => Ord valueWrapper
+    => NormalizedAc key valueWrapper child
+    -> Maybe child
+asSingleOpaqueElem
+    NormalizedAc
+        { elementsWithVariables
+        , concreteElements
+        , opaque
+        }
+    | elementsWithVariables == mempty
```

Comment:
I would suggest `null` instead of `== mempty`. You can also use a pattern guard to avoid repeating the `Nothing`:

```suggestion
    | null elementsWithVariables
    , null concreteElements 
    , [singleOpaqueElem] <- opaque
    = Just singleOpaqueElem
    | otherwise = Nothing
```